imap/tls.c

```diff
@@ -1093,13 +1093,13 @@ static long bio_dump_cb(BIO * bio, int cmd, const char *argp, int argi,
         return (ret);
 
     if (cmd == (BIO_CB_READ | BIO_CB_RETURN)) {
-        printf("read from %08lX [%08lX] (%d bytes => %ld (0x%lX))",
+        printf("read from %0lX [%0lX] (%d bytes => %ld (0x%lX))",
```

Comment:
I agree this code is broken, but this doesn't fix it.  The problem is the pointer arguments have been cast to `unsigned long` on the next line, which might be smaller than the pointer type.

It should be using %p instead of %X for the pointer arguments, and should not be casting them to `unsigned long`.

Reply:
indeed.